x/twap/types/genesis_test.go

```diff
@@ -218,12 +253,57 @@ func TestTWAPRecord_Validate(t *testing.T) {
 
 			expectedErr: true,
 		},
-	}
+		"invalid p1 arithmetic accum: negative": {
+			twapRecord: func() TwapRecord {
+				r := baseRecord
+				r.P1ArithmeticTwapAccumulator = sdk.OneDec().Neg()
+				return r
+			}(),
+
+			expectedErr: true,
+		},
+		"invalid p0 arithmetic accum: nil": {
+			twapRecord: func() TwapRecord {
+				r := TwapRecord{
+					PoolId:                      basePoolId,
+					Asset0Denom:                 denom0,
+					Asset1Denom:                 denom1,
+					Height:                      3,
+					Time:                        tPlusOne.Add(time.Second),
+					P0LastSpotPrice:             sdk.OneDec(),
+					P1LastSpotPrice:             sdk.OneDec(),
+					P1ArithmeticTwapAccumulator: sdk.OneDec(),
+				}
+				return r
+			}(),
 
+			expectedErr: true,
+		},
+		"invalid p1 arithmetic accum: nil": {
+			twapRecord: func() TwapRecord {
+				r := TwapRecord{
+					PoolId:                      basePoolId,
+					Asset0Denom:                 denom0,
+					Asset1Denom:                 denom1,
+					Height:                      3,
+					Time:                        tPlusOne.Add(time.Second),
+					P0LastSpotPrice:             sdk.OneDec(),
+					P1LastSpotPrice:             sdk.OneDec(),
+					P0ArithmeticTwapAccumulator: sdk.OneDec(),
+				}
+				return r
+			}(),
+			expectedErr: true,
+		},
+	}
```

Comment:
Should be handled by the for loop before to make it symmetric

Reply:
The symmetric loop doesn't work for nil values, so that is why I duplicated them for both sides.